[2025-03-06T16:02:39.962+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-03-06T16:02:39.989+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_flow.load_exchangeRate_python_job manual__2025-03-06T16:02:36.507775+00:00 [queued]>
[2025-03-06T16:02:40.001+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_flow.load_exchangeRate_python_job manual__2025-03-06T16:02:36.507775+00:00 [queued]>
[2025-03-06T16:02:40.002+0000] {taskinstance.py:2867} INFO - Starting attempt 1 of 1
[2025-03-06T16:02:40.017+0000] {taskinstance.py:2890} INFO - Executing <Task(SparkSubmitOperator): load_exchangeRate_python_job> on 2025-03-06 16:02:36.507775+00:00
[2025-03-06T16:02:40.024+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=2966) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-03-06T16:02:40.025+0000] {standard_task_runner.py:72} INFO - Started process 2967 to run task
[2025-03-06T16:02:40.024+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'spark_flow', 'load_exchangeRate_python_job', 'manual__2025-03-06T16:02:36.507775+00:00', '--job-id', '40', '--raw', '--subdir', 'DAGS_FOLDER/dag_***_spark.py', '--cfg-path', '/tmp/tmpnhgt3uor']
[2025-03-06T16:02:40.027+0000] {standard_task_runner.py:105} INFO - Job 40: Subtask load_exchangeRate_python_job
[2025-03-06T16:02:40.087+0000] {task_command.py:467} INFO - Running <TaskInstance: spark_flow.load_exchangeRate_python_job manual__2025-03-06T16:02:36.507775+00:00 [running]> on host e9dfa152b6ab
[2025-03-06T16:02:40.186+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='haonv' AIRFLOW_CTX_DAG_ID='spark_flow' AIRFLOW_CTX_TASK_ID='load_exchangeRate_python_job' AIRFLOW_CTX_EXECUTION_DATE='2025-03-06T16:02:36.507775+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-03-06T16:02:36.507775+00:00'
[2025-03-06T16:02:40.187+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-03-06T16:02:40.216+0000] {base.py:84} INFO - Retrieving connection 'spark-conn'
[2025-03-06T16:02:40.218+0000] {spark_submit.py:474} INFO - Spark-Submit cmd: spark-submit --master spark://172.18.0.4:7077 --conf spark.jars=file:///opt/bitnami/spark/jars/postgresql-42.7.5.jar --name arrow-spark --deploy-mode client jobs/python/load_exchangeRateToDW.py
[2025-03-06T16:02:41.867+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:41 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-03-06T16:02:42.027+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:42 WARN DependencyUtils: Local jar /opt/bitnami/spark/jars/postgresql-42.7.5.jar does not exist, skipping.
[2025-03-06T16:02:43.502+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:43 INFO SparkContext: Running Spark version 3.5.5
[2025-03-06T16:02:43.503+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:43 INFO SparkContext: OS info Linux, 5.15.167.4-microsoft-standard-WSL2, amd64
[2025-03-06T16:02:43.504+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:43 INFO SparkContext: Java version 17.0.14
[2025-03-06T16:02:43.534+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:43 INFO ResourceUtils: ==============================================================
[2025-03-06T16:02:43.534+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:43 INFO ResourceUtils: No custom resources configured for spark.driver.
[2025-03-06T16:02:43.535+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:43 INFO ResourceUtils: ==============================================================
[2025-03-06T16:02:43.535+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:43 INFO SparkContext: Submitted application: ExchangeRateToDW
[2025-03-06T16:02:43.555+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:43 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2025-03-06T16:02:43.563+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:43 INFO ResourceProfile: Limiting resource is cpu
[2025-03-06T16:02:43.564+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:43 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2025-03-06T16:02:43.613+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:43 INFO SecurityManager: Changing view acls to: ***
[2025-03-06T16:02:43.614+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:43 INFO SecurityManager: Changing modify acls to: ***
[2025-03-06T16:02:43.615+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:43 INFO SecurityManager: Changing view acls groups to:
[2025-03-06T16:02:43.615+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:43 INFO SecurityManager: Changing modify acls groups to:
[2025-03-06T16:02:43.616+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:43 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: ***; groups with view permissions: EMPTY; users with modify permissions: ***; groups with modify permissions: EMPTY
[2025-03-06T16:02:43.821+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:43 INFO Utils: Successfully started service 'sparkDriver' on port 46443.
[2025-03-06T16:02:43.844+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:43 INFO SparkEnv: Registering MapOutputTracker
[2025-03-06T16:02:43.875+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:43 INFO SparkEnv: Registering BlockManagerMaster
[2025-03-06T16:02:43.890+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:43 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2025-03-06T16:02:43.891+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:43 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2025-03-06T16:02:43.895+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:43 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2025-03-06T16:02:43.925+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:43 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-911590a7-35c9-4440-ac4c-e139c33f7f81
[2025-03-06T16:02:43.937+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:43 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2025-03-06T16:02:43.952+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:43 INFO SparkEnv: Registering OutputCommitCoordinator
[2025-03-06T16:02:44.067+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:44 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2025-03-06T16:02:44.145+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:44 INFO Utils: Successfully started service 'SparkUI' on port 4040.
[2025-03-06T16:02:44.184+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:44 ERROR SparkContext: Failed to add file:///opt/bitnami/spark/jars/postgresql-42.7.5.jar to Spark environment
[2025-03-06T16:02:44.184+0000] {spark_submit.py:641} INFO - java.io.FileNotFoundException: Jar /opt/bitnami/spark/jars/postgresql-42.7.5.jar not found
[2025-03-06T16:02:44.185+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.addLocalJarFile$1(SparkContext.scala:2095)
[2025-03-06T16:02:44.186+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.addJar(SparkContext.scala:2151)
[2025-03-06T16:02:44.186+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.$anonfun$new$15(SparkContext.scala:521)
[2025-03-06T16:02:44.187+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.$anonfun$new$15$adapted(SparkContext.scala:521)
[2025-03-06T16:02:44.187+0000] {spark_submit.py:641} INFO - at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2025-03-06T16:02:44.188+0000] {spark_submit.py:641} INFO - at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2025-03-06T16:02:44.188+0000] {spark_submit.py:641} INFO - at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2025-03-06T16:02:44.189+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.<init>(SparkContext.scala:521)
[2025-03-06T16:02:44.189+0000] {spark_submit.py:641} INFO - at org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
[2025-03-06T16:02:44.190+0000] {spark_submit.py:641} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2025-03-06T16:02:44.190+0000] {spark_submit.py:641} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
[2025-03-06T16:02:44.191+0000] {spark_submit.py:641} INFO - at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2025-03-06T16:02:44.192+0000] {spark_submit.py:641} INFO - at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
[2025-03-06T16:02:44.192+0000] {spark_submit.py:641} INFO - at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
[2025-03-06T16:02:44.193+0000] {spark_submit.py:641} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
[2025-03-06T16:02:44.193+0000] {spark_submit.py:641} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2025-03-06T16:02:44.194+0000] {spark_submit.py:641} INFO - at py4j.Gateway.invoke(Gateway.java:238)
[2025-03-06T16:02:44.194+0000] {spark_submit.py:641} INFO - at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
[2025-03-06T16:02:44.195+0000] {spark_submit.py:641} INFO - at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
[2025-03-06T16:02:44.195+0000] {spark_submit.py:641} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2025-03-06T16:02:44.196+0000] {spark_submit.py:641} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2025-03-06T16:02:44.196+0000] {spark_submit.py:641} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2025-03-06T16:02:44.253+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:44 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://172.18.0.4:7077...
[2025-03-06T16:02:44.289+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:44 INFO TransportClientFactory: Successfully created connection to /172.18.0.4:7077 after 20 ms (0 ms spent in bootstraps)
[2025-03-06T16:02:44.363+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:44 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20250306160244-0010
[2025-03-06T16:02:44.366+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:44 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20250306160244-0010/0 on worker-20250306150809-172.18.0.7-35707 (172.18.0.7:35707) with 2 core(s)
[2025-03-06T16:02:44.369+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:44 INFO StandaloneSchedulerBackend: Granted executor ID app-20250306160244-0010/0 on hostPort 172.18.0.7:35707 with 2 core(s), 1024.0 MiB RAM
[2025-03-06T16:02:44.370+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:44 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20250306160244-0010/1 on worker-20250306150810-172.18.0.10-33947 (172.18.0.10:33947) with 2 core(s)
[2025-03-06T16:02:44.370+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:44 INFO StandaloneSchedulerBackend: Granted executor ID app-20250306160244-0010/1 on hostPort 172.18.0.10:33947 with 2 core(s), 1024.0 MiB RAM
[2025-03-06T16:02:44.374+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:44 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 33635.
[2025-03-06T16:02:44.375+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:44 INFO NettyBlockTransferService: Server created on e9dfa152b6ab:33635
[2025-03-06T16:02:44.378+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:44 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2025-03-06T16:02:44.387+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:44 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, e9dfa152b6ab, 33635, None)
[2025-03-06T16:02:44.391+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:44 INFO BlockManagerMasterEndpoint: Registering block manager e9dfa152b6ab:33635 with 434.4 MiB RAM, BlockManagerId(driver, e9dfa152b6ab, 33635, None)
[2025-03-06T16:02:44.395+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:44 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, e9dfa152b6ab, 33635, None)
[2025-03-06T16:02:44.397+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:44 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, e9dfa152b6ab, 33635, None)
[2025-03-06T16:02:44.398+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:44 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20250306160244-0010/0 is now RUNNING
[2025-03-06T16:02:44.400+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:44 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20250306160244-0010/1 is now RUNNING
[2025-03-06T16:02:44.683+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:44 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2025-03-06T16:02:44.955+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:44 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2025-03-06T16:02:44.958+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:44 INFO SharedState: Warehouse path is 'file:/opt/***/spark-warehouse'.
[2025-03-06T16:02:46.689+0000] {spark_submit.py:641} INFO - Traceback (most recent call last):
[2025-03-06T16:02:46.699+0000] {spark_submit.py:641} INFO - File "/opt/***/jobs/python/load_exchangeRateToDW.py", line 65, in <module>
[2025-03-06T16:02:46.709+0000] {spark_submit.py:641} INFO - .load()
[2025-03-06T16:02:46.710+0000] {spark_submit.py:641} INFO - ^^^^^^
[2025-03-06T16:02:46.711+0000] {spark_submit.py:641} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/streaming/readwriter.py", line 304, in load
[2025-03-06T16:02:46.712+0000] {spark_submit.py:641} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py", line 1322, in __call__
[2025-03-06T16:02:46.713+0000] {spark_submit.py:641} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/pyspark.zip/pyspark/errors/exceptions/captured.py", line 185, in deco
[2025-03-06T16:02:46.719+0000] {spark_submit.py:641} INFO - pyspark.errors.exceptions.captured.AnalysisException: Failed to find data source: kafka. Please deploy the application as per the deployment section of Structured Streaming + Kafka Integration Guide.
[2025-03-06T16:02:46.824+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:46 INFO SparkContext: Invoking stop() from shutdown hook
[2025-03-06T16:02:46.829+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:46 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2025-03-06T16:02:46.849+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:46 INFO SparkUI: Stopped Spark web UI at http://e9dfa152b6ab:4040
[2025-03-06T16:02:46.857+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:46 INFO StandaloneSchedulerBackend: Shutting down all executors
[2025-03-06T16:02:46.864+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:46 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Asking each executor to shut down
[2025-03-06T16:02:46.893+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:46 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2025-03-06T16:02:46.909+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:46 INFO MemoryStore: MemoryStore cleared
[2025-03-06T16:02:46.910+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:46 INFO BlockManager: BlockManager stopped
[2025-03-06T16:02:46.917+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:46 INFO BlockManagerMaster: BlockManagerMaster stopped
[2025-03-06T16:02:46.922+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:46 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2025-03-06T16:02:46.932+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:46 INFO SparkContext: Successfully stopped SparkContext
[2025-03-06T16:02:46.933+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:46 INFO ShutdownHookManager: Shutdown hook called
[2025-03-06T16:02:46.933+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:46 INFO ShutdownHookManager: Deleting directory /tmp/spark-89b1d821-40ea-4c29-af00-d6f0f4bd6009
[2025-03-06T16:02:46.937+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:46 INFO ShutdownHookManager: Deleting directory /tmp/spark-76852bf7-0cb5-452d-97d9-ba255ae696a9
[2025-03-06T16:02:46.942+0000] {spark_submit.py:641} INFO - 25/03/06 16:02:46 INFO ShutdownHookManager: Deleting directory /tmp/spark-76852bf7-0cb5-452d-97d9-ba255ae696a9/pyspark-834133a4-a78e-4cde-b9d2-1eb32f398b8d
[2025-03-06T16:02:46.994+0000] {taskinstance.py:3313} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 177, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 567, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://172.18.0.4:7077 --conf spark.jars=file:///opt/bitnami/spark/jars/postgresql-42.7.5.jar --name arrow-spark --deploy-mode client jobs/python/load_exchangeRateToDW.py. Error code is: 1.
[2025-03-06T16:02:46.998+0000] {taskinstance.py:1226} INFO - Marking task as FAILED. dag_id=spark_flow, task_id=load_exchangeRate_python_job, run_id=manual__2025-03-06T16:02:36.507775+00:00, execution_date=20250306T160236, start_date=20250306T160239, end_date=20250306T160246
[2025-03-06T16:02:47.031+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-03-06T16:02:47.032+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 40 for task load_exchangeRate_python_job (Cannot execute: spark-submit --master spark://172.18.0.4:7077 --conf spark.jars=file:///opt/bitnami/spark/jars/postgresql-42.7.5.jar --name arrow-spark --deploy-mode client jobs/python/load_exchangeRateToDW.py. Error code is: 1.; 2967)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3006, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 274, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3161, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3185, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 177, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 567, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://172.18.0.4:7077 --conf spark.jars=file:///opt/bitnami/spark/jars/postgresql-42.7.5.jar --name arrow-spark --deploy-mode client jobs/python/load_exchangeRateToDW.py. Error code is: 1.
[2025-03-06T16:02:47.077+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-03-06T16:02:47.105+0000] {taskinstance.py:3901} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-03-06T16:02:47.113+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
